### PR TITLE
Add army support for Lumineth Realm-Lords

### PR DESCRIPTION
**Pull request by amcalvert**

### CODE REVIEW

File: src/army/lumineth/allegiances.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
+import { TAllegiances } from 'types/army'
+import { DURING_GAME, HERO_PHASE, MOVEMENT_PHASE, COMBAT_PHASE, WOUND_ALLOCATION, SHOOTING_PHASE, END_OF_COMBAT_PHASE, START_OF_COMBAT_PHASE } from 'types/phases'
+
+const Allegiances: TAllegiances = [
+  {
+    name: `Ymetrica`,
+    effects: [
+      {
+        name: `Mountain Realm`,
+        desc: `The Enduring as Rock battle trait chages an attack that targets a YMETRICA ALARITH unit in mountain stand to '-' if ` +
```
Comment (exonian) [inline src/army/lumineth/allegiances.ts:10]: typo `chages` = `changes`

File: src/army/lumineth/allegiances.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        desc: `Pick 1 of the bearer's weapons, once per phase add 1 damage inflicted by 1 attack. ` +
+              `Additionally, roll a dice each time you allocate a wound or mortal wound to the bearer, on a 6+ it is negated. ` + 
+              `Additionally, roll a dice each time the bearer is inflicted by a spell or endless spell, on a 5+ ignore the effects.`,
+        when: [COMBAT_PHASE, WOUND_ALLOCATION, HERO_PHASE],
+        artifact: true
+      },
+      {
+        name: `Redoubled Force`,
+        desc: `Pick 1 YMETRICA ALARITH unit that has forced an enemy unit to move using the Tectonic Force battle trait for the first time ` +
+              `in that phase and that is wholly within 18" of a friendly YMETRICA HERO. You can use Tectonia Force second time by picking ` +
```
Comment (exonian) [inline src/army/lumineth/allegiances.ts:25]: `Tectonic`

File: src/army/lumineth/allegiances.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        name: `Deplete Reserves`,
+        desc: `You can use this ability when a friendly SYAR unit would use its aetherquartz reserve ability, even if any friendly ` + 
+              `SYAR units have already done so in that turn. Pick 1 friendly SYAR unit that has any aetherquartz reserves and is ` + 
+              `wholly within 18" of a friendly SYAR HERO. That unit can use one of its aetherquartz reserves.`,
+        when: [DURING_GAME],
+        command_ability: true,
+      },
+      {
+        name: `Goading Arrogance`,
+        desc: `You can pick 1 enemy HERO withn 6" of this general. That enemy HERO can only target this general in that phase. ` +
```
Comment (exonian) [inline src/army/lumineth/allegiances.ts:62]: `within`

File: src/army/lumineth/allegiances.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+      {
+        name: `Fast Learner`,
+        desc: `This general can attrmpt to unbind 1 extra spell in the enemy hero phase. In addition, the second time that this ` +
+              `general attempts to unbind a spell in the same enemy hero phase, you can re-roll the unbinding roll.`,
+        when: [HERO_PHASE],
+        command_trait: true,
+      },
+      {
+        name: `Gift of Celennar`,
+        desc: `Roll a dice each time you allocate a wound or mortal wound (add 2 to the roll if TECLIS is part of your army and on the battlefield). On a 6+ the wound or mortal wound is negated`,
```
Comment (exonian) [inline src/army/lumineth/allegiances.ts:117]: Is this definitely keyword TECLIS?

File: src/army/lumineth/spells.ts [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        name: `Assault of Stone`,
+        desc: `Casting value of 8. Pick one enemy model within 24" of the caster and visible to them. ` +
+              `Roll a number of dice equal to the casting roll. For each roll that is less than the units Save characteristic, that unit suffers 1 mortal wound. ` +
+              `Rolls of 1 or 2 always fail. A Save characteristic of '-' counts as a 6.`,
+        when: [HERO_PHASE],
+      },
+    ],
+  },
+  {
+    name: `Assault of Stone`,
```
Comment (exonian) [inline src/army/lumineth/spells.ts:144]: Think this is a copy-paste duplicate

File: src/army/lumineth/allegiances.ts (hunk 1 above)
Comment (Flui) [inline src/army/lumineth/allegiances.ts:10]: There is no need to concatenate those strings with a `+` as Template-strings can be written multiline.

e.g.
```javasctipt
desc: `The Enduring as Rock battle trait chages an attack that targets a YMETRICA ALARITH unit in mountain stand to '-' if 
the weapon used for the attack has Rend -1 or -2.`,
```

File: src/army/lumineth/allegiances.ts [hunk 6]
```diff
@@ -0,0 +1,133 @@
+import { TAllegiances } from 'types/army'
+import { DURING_GAME, HERO_PHASE, MOVEMENT_PHASE, COMBAT_PHASE, WOUND_ALLOCATION, SHOOTING_PHASE, END_OF_COMBAT_PHASE, START_OF_COMBAT_PHASE } from 'types/phases'
```
Comment (Flui) [inline src/army/lumineth/allegiances.ts:2]: This exceeds the maximum line-width defined in the prettier config:  https://github.com/daviseford/aos-reminders/blob/2021f4a376e181295ee4de542e2c4d5238a14c8b/.prettierrc#L6.

If you run `yarn format` this should be fixed

Comment (Flui) [reply src/army/lumineth/allegiances.ts:2]: this seems to happen multiple times

File: src/army/lumineth/index.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+import Allegiances from './allegiances'
+import Artifacts from './artifacts'
+import EndlessSpells from './endless_spells'
+import Spells from './spells'
+import Traits from './traits'
+
+export default {
+  Abilities,
+  Allegiances,
+  AllegianceType: 'Great Nations',
```
Comment (Flui) [inline src/army/lumineth/index.ts:12]: what would you think of putting this in an extra const between the import and exports?

File: src/army/lumineth/allegiances.ts (hunk 1 above)
Comment (amcalvert) [reply src/army/lumineth/allegiances.ts:10]: Renamed

File: src/army/lumineth/allegiances.ts (hunk 3 above)
Comment (amcalvert) [reply src/army/lumineth/allegiances.ts:62]: Updated

File: src/army/lumineth/allegiances.ts (hunk 2 above)
Comment (amcalvert) [reply src/army/lumineth/allegiances.ts:25]: Updated

File: src/army/lumineth/allegiances.ts (hunk 4 above)
Comment (amcalvert) [reply src/army/lumineth/allegiances.ts:117]: Yep

File: src/army/lumineth/spells.ts (hunk 5 above)
Comment (amcalvert) [reply src/army/lumineth/spells.ts:144]: Yeah not sure how i managed that, removed

File: src/army/lumineth/allegiances.ts (hunk 1 above)
Comment (amcalvert) [reply src/army/lumineth/allegiances.ts:10]: I didn't know that :) Don't write enough JS/TS these days. Updated

File: src/army/lumineth/allegiances.ts (hunk 6 above)
Comment (amcalvert) [reply src/army/lumineth/allegiances.ts]: Good call, done

File: src/army/lumineth/index.ts (hunk 7 above)
Comment (amcalvert) [reply src/army/lumineth/index.ts]: Consistency wise other index.ts files don't (example Kharadron) but have implemented as tidier.

File: src/army/lumineth/allegiances.ts (hunk 6 above)
Comment (Flui) [reply src/army/lumineth/allegiances.ts]: there is just one problem I think that prettier does not format strings, so you might have to do it on your own

File: src/army/lumineth/index.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import { AlliedUnits, Battalions, Units } from './units'
+import Abilities from './abilities'
+import Allegiances from './allegiances'
+import Artifacts from './artifacts'
+import EndlessSpells from './endless_spells'
+import Spells from './spells'
+import Traits from './traits'
+
+const LuminethAlligenceType = {
+  AllegianceType: 'Great Nations',
```
Comment (Flui) [inline src/army/lumineth/index.ts:10]: I dont think that this works as expected as AllegienceType seems to be a mandatory attribute of an army.

```javascript
const AllegienceType = 'Great Nations';

export default {
  ...,
  AllegienceType,
  ...,
}
```

File: src/army/lumineth/spells.ts [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+      {
+        name: `Lambent Light`,
+        desc: `Casting value of 8. Pick 1 enemy unit wholly within 18" and visible to them. 
+              Until your next hero phase, re-roll missile weapon attacks that target the unit.`,
+        when: [HERO_PHASE, SHOOTING_PHASE],
+      },
+    ],
+  },
+  {
+    name: `Ethreal Blessing`,
```
Comment (exonian) [inline src/army/lumineth/spells.ts:49]: Typo

Comment (amcalvert) [reply src/army/lumineth/spells.ts]: Ah spelling, why hast thou forsaken me

File: src/army/lumineth/index.ts (hunk 8 above)
Comment (amcalvert) [reply src/army/lumineth/index.ts]: Transpiles oddly enough but will cause problems downstream i assume, changed.

File: src/army/lumineth/allegiances.ts [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+    effects: [
+      {
+        name: `Lambent Mystics`,
+        desc: `Add 1 to the first casting, dispelling or unbinding roll you make for each ZAITREC WIZARD in each hero phase. 
+              In addition, each ZAITREC WIZARD HERO knows 1 extra spell form the appropriate lore`,
+        when: [HERO_PHASE],
+      },
+      {
+        name: `Fast Learner`,
+        desc: `This general can attrmpt to unbind 1 extra spell in the enemy hero phase. In addition, the second time that this
```
Comment (daviseford) [inline src/army/lumineth/allegiances.ts:107]: attrmpt -> attempt

File: src/army/lumineth/commonUnitEffects.ts [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+
+export const getVanariWizardsEffect = (minimumModelCountToBeWizard: number) => {
+  return {
+    name: `Magic`,
+    desc: `This unit is a WIZARD while it has ${minimumModelCountToBeWizard} or more models. They can attempt to cast 1 spell in your hero phase and attempt to unbind 1 spell in the enemy hero phase. They know the Power of Hysh spell.`,
+    when: [HERO_PHASE],
+  }
+}
+
+export const alarithSpiritFreeCommandAbilityEffect = (effectName: string, effectRange: number) => {
```
Comment (daviseford) [inline src/army/lumineth/commonUnitEffects.ts:27]: Nice work with these!

File: src/army/lumineth/allegiances.ts (hunk 10 above)
Comment (amcalvert) [reply src/army/lumineth/allegiances.ts]: Updated

File: src/army/lumineth/commonUnitEffects.ts (hunk 11 above)
Comment (amcalvert) [reply src/army/lumineth/commonUnitEffects.ts:27]: Thanks, was just riffing off what @exonian started there so kudos over there too.

**Comment by amcalvert** [comment]
All changes made, just gonna go through it with some dev-testing then I'm done.

File: src/army/lumineth/commonUnitEffects.ts (hunk 11 above)
Comment (exonian) [reply src/army/lumineth/commonUnitEffects.ts:27]: Thanks! In turn, I picked it up from a couple of things @daviseford had done to tidy things up. Feels good to collaborate to make some solid patterns here that should likely make for more reliable data.

**Comment by amcalvert** [comment]
Done from my side

**Pull request merged**